framework/include/mfem/executioners/MFEMExecutioner.h

```diff
@@ -0,0 +1,33 @@
+#ifdef MFEM_ENABLED
+
+#pragma once
+#include "Executioner.h"
+#include "MFEMProblemData.h"
+
+class MFEMProblem;
+
+class MFEMExecutioner : public Executioner
+{
+public:
+  static InputParameters validParams();
+
+  MFEMExecutioner(const InputParameters & params);
+  virtual ~MFEMExecutioner() override = default;
+
+  virtual bool lastSolveConverged() const override { return true; };
+
+  /// Virtual method to construct the ProblemOperator. Call for default problems.
+  virtual void constructProblemOperator() = 0;
+
+  /**
+   * Set the device to use to solve the FE problem.
+   */
+  void setDevice();
```

Comment:
We should have a discussion at some point on devices and how to generalize specifying something like a GPU in moose